In [1]:
import yaml
import time
from pandas import json_normalize
import pandas as pd
import xlrd

In [4]:
key_dict = yaml.load(open('../../../Credentials/credentialsGoogle.yml'), Loader=yaml.FullLoader)
gmaps_key = key_dict['key']

In [176]:
import requests

url = "https://maps.googleapis.com/maps/api/place/textsearch/json?"
query = 'supermarket near Zurich'

api_key = gmaps_key

pagetoken="*"
count=1
all_result={}


counter = 1

all_results = {}

next_page_token = 'START'

count =1 
pagetoken="*"

search_string=['Migros','Migrolino','Lidl','Aldi','Coop Supermarkt','Coop Pronto','Coop City']
query_end=" near Zurich"
query_list = [shop+query_end for shop in search_string]
print(query_list)
for query in query_list:
    pagetoken = "*"
    print("ahead of while loop")
    count_while=1
    while(pagetoken!=""):
        full_query = url + 'query=' + query + '&key=' + api_key   
        if count_while > 1:
             full_query += '&pagetoken=' + pagetoken
        print(full_query)
        req = requests.get(full_query) 
        places_json = req.json() 
    
        if places_json['status'] == 'INVALID_REQUEST':
            print("in break point")
            break
        all_result[count] = places_json['results'] 
        print(all_result[count][0])
        count_while+=1
        count+=1
        
        try:
            print(count,pagetoken)
            pagetoken = places_json['next_page_token']
            #print(count+1,pagetoken)
        except:
            pagetoken = ""
        time.sleep(1.5)

print(count)


  
# now result contains list of nested dictionaries 
#my_result = places_json['results'] 
#print("next",my_result[0])
# take a look at the first element

#name
#geometry lat/long
#place_id
#rating
#userrating
#types
#address

['Migros near Zurich', 'Migrolino near Zurich', 'Lidl near Zurich', 'Aldi near Zurich', 'Coop Supermarkt near Zurich', 'Coop Pronto near Zurich', 'Coop City near Zurich']
ahead of while loop
https://maps.googleapis.com/maps/api/place/textsearch/json?query=Migros near Zurich&key=AIzaSyAqh4ivdn4JKv8ShX2Z9IZeUrIXZmEGYN4
{'business_status': 'OPERATIONAL', 'formatted_address': 'Löwenstrasse 31-35, 8001 Zürich, Switzerland', 'geometry': {'location': {'lat': 47.3753395, 'lng': 8.5363169}, 'viewport': {'northeast': {'lat': 47.37663087989272, 'lng': 8.537792329892723}, 'southwest': {'lat': 47.37393122010728, 'lng': 8.535092670107279}}}, 'icon': 'https://maps.gstatic.com/mapfiles/place_api/icons/shopping-71.png', 'name': 'Migros', 'opening_hours': {'open_now': True}, 'photos': [{'height': 2976, 'html_attributions': ['<a href="https://maps.google.com/maps/contrib/100125680457952180690">Ruzic Violeta</a>'], 'photo_reference': 'CmRaAAAA1vex82qrJk7aInAvtnzVcY7M4zNQ9qHaahsaQKlv_PzhjWc1QX6Ybgh-cv-YI-X

ahead of while loop
https://maps.googleapis.com/maps/api/place/textsearch/json?query=Aldi near Zurich&key=AIzaSyAqh4ivdn4JKv8ShX2Z9IZeUrIXZmEGYN4
{'business_status': 'OPERATIONAL', 'formatted_address': 'Zollstrasse 58, 8005 Zürich, Switzerland', 'geometry': {'location': {'lat': 47.3809072, 'lng': 8.5332134}, 'viewport': {'northeast': {'lat': 47.38217532989272, 'lng': 8.53451642989272}, 'southwest': {'lat': 47.37947567010728, 'lng': 8.531816770107277}}}, 'icon': 'https://maps.gstatic.com/mapfiles/place_api/icons/shopping-71.png', 'name': 'Aldi', 'opening_hours': {'open_now': True}, 'photos': [{'height': 1192, 'html_attributions': ['<a href="https://maps.google.com/maps/contrib/110553906027621935592">A Google User</a>'], 'photo_reference': 'CmRaAAAAGcePo9KH8ga0IOBMkM1vS1rNQESbhCt6NSjwmxrpucatjdy6J_mf1BnBxXkDtCdvLmeZqDSqR_SAYWnyNJAi0szF-2fA4uBhfUqUmE4rDJkeS-WOwW-EMGcpbScvJEOSEhAQsNj8qZlaGNpVqjQmYN2sGhRBnDshfqxILuT5ccGPRoaouIokOw', 'width': 2119}], 'place_id': 'ChIJDXAJGA4KkEcRwgtNwO0k-og'

ahead of while loop
https://maps.googleapis.com/maps/api/place/textsearch/json?query=Coop City near Zurich&key=AIzaSyAqh4ivdn4JKv8ShX2Z9IZeUrIXZmEGYN4
{'business_status': 'OPERATIONAL', 'formatted_address': 'Kalanderpl. 1, 8045 Zürich, Switzerland', 'geometry': {'location': {'lat': 47.3574543, 'lng': 8.5228465}, 'viewport': {'northeast': {'lat': 47.35879687989272, 'lng': 8.52419762989272}, 'southwest': {'lat': 47.35609722010727, 'lng': 8.521497970107276}}}, 'icon': 'https://maps.gstatic.com/mapfiles/place_api/icons/shopping-71.png', 'name': 'Coop City', 'opening_hours': {'open_now': True}, 'photos': [{'height': 3000, 'html_attributions': ['<a href="https://maps.google.com/maps/contrib/115809073337729603059">Donat Deger</a>'], 'photo_reference': 'CmRaAAAAdpkDrIZZsEvCxJT-DQtzX4SmxiyEcDzJfRrTnwWAsEFOgvHpRoueMh0F2EhDScpFZt2TyI677DlWUHxGWkW_btup3IkIVjWiHQsplrGoVRLyRHKkS9DZN_TqhJH8XG7KEhA9M3bHZa071SGzYSUJqMBAGhTG5T1h2KsHjbJykQ5vAhfkrcBovw', 'width': 4000}], 'place_id': 'ChIJp3j71fEJkEcRBzbWC

In [226]:
df_stores = pd.DataFrame({'A' : []})
first_loop=True
for key,value in all_result.items():
    if first_loop:
        df_stores=json_normalize(value)
        first_loop=False
    else:
        df_stores=pd.concat([df_stores, json_normalize(value)], axis=0)
print(df_stores.shape)
df_stores.head()          

(195, 21)


business_status                                  formatted_address  \
0     OPERATIONAL       Löwenstrasse 31-35, 8001 Zürich, Switzerland   
1     OPERATIONAL             Hauptbahnhof, 8001 Zürich, Switzerland   
2     OPERATIONAL  Schaffhauserstrasse 470, 8052 Zürich, Switzerland   
3     OPERATIONAL        Limmatstrasse 152, 8005 Zürich, Switzerland   
4     OPERATIONAL  Birmensdorferstrasse 320, 8055 Zürich, Switzer...   

                                                icon               name  \
0  https://maps.gstatic.com/mapfiles/place_api/ic...             Migros   
1  https://maps.gstatic.com/mapfiles/place_api/ic...  Migros Supermarkt   
2  https://maps.gstatic.com/mapfiles/place_api/ic...  Migros Supermarkt   
3  https://maps.gstatic.com/mapfiles/place_api/ic...  Migros Supermarkt   
4  https://maps.gstatic.com/mapfiles/place_api/ic...  Migros Supermarkt   

                                              photos  \
0  [{'height': 2976, 'html_attributions': ['<a hr...   
1  [{'height': 1960, 'html_attributions': ['<a hr...   
2  [{'height': 4640, 'html_attributions': ['<a hr...   
3  [{'height': 1836, 'html_attributions': ['<a hr...   
4  [{'height': 2204, 'html_attributions': ['<a hr...   

                      place_id  rating                    reference  \
0  ChIJebvE0AUKkEcRli-dLu8vFaM     4.3  ChIJebvE0AUKkEcRli-dLu8vFaM   
1  ChIJS6CT8QgKkEcRwqhtQkVOTUI     4.2  ChIJS6CT8QgKkEcRwqhtQkVOTUI   
2  ChIJzcr5wpsKkEcR8OcT8pamnyI     4.3  ChIJzcr5wpsKkEcR8OcT8pamnyI   
3  ChIJS6CT8QgKkEcRVC558wYywsQ     4.3  ChIJS6CT8QgKkEcRVC558wYywsQ   
4  ChIJlyh8cScKkEcRY3M2wmnjbVI     4.3  ChIJlyh8cScKkEcRY3M2wmnjbVI   

                                               types  user_ratings_total  ...  \
0  [grocery_or_supermarket, supermarket, food, po...                1754  ...   
1  [grocery_or_supermarket, supermarket, food, po...                1246  ...   
2  [grocery_or_supermarket, supermarket, food, po...                 532  ...   
3  [grocery_or_supermarket, supermarket, food, po...                1266  ...   
4  [grocery_or_supermarket, supermarket, food, po...                 414  ...   

   geometry.location.lng  geometry.viewport.northeast.lat  \
0               8.536317                        47.376631   
1               8.540276                        47.379774   
2               8.548329                        47.420927   
3               8.531396                        47.386682   
4               8.508698                        47.371234   

   geometry.viewport.northeast.lng  geometry.viewport.southwest.lat  \
0                         8.537792                        47.373931   
1                         8.541851                        47.377074   
2                         8.549585                        47.418227   
3                         8.532517                        47.383983   
4                         8.510071                        47.368534   

   geometry.viewport.southwest.lng  opening_hours.open_now  \
0                         8.535093                    True   
1                         8.539151                    True   
2                         8.546886                    True   
3                         8.529818                    True   
4                         8.507371                    True   

       plus_code.compound_code plus_code.global_code permanently_closed  \
0  9GGP+4G Zürich, Switzerland           8FVC9GGP+4G                NaN   
1  9GHR+84 Zürich, Switzerland           8FVC9GHR+84                NaN   
2  CG9X+R8 Zürich, Switzerland           8FVCCG9X+R8                NaN   
3  9GPJ+5H Zürich, Switzerland           8FVC9GPJ+5H                NaN   
4  9G95+XF Zürich, Switzerland           8FVC9G95+XF                NaN   

  price_level  
0         NaN  
1         NaN  
2         NaN  
3         NaN  
4         NaN  

[5 rows x 21 columns]

In [227]:
df_stores[['PostalCode']] = df_stores["formatted_address"].str.extract("(\d\d\d\d)",expand=True)

In [228]:
search_string=['Migros','Migrolino','Lidl','Aldi','Coop Supermarkt','Coop Pronto','Coop City']
query_end=" near Zurich"

In [229]:
df_stores[['Street','city country']] = df_stores["formatted_address"].str.split(", \d\d\d\d ",expand=True)
print(df_stores.isna().sum())
df_stores.head()

business_status                      0
formatted_address                    0
icon                                 0
name                                 0
photos                               7
place_id                             0
rating                               0
reference                            0
types                                0
user_ratings_total                   0
geometry.location.lat                0
geometry.location.lng                0
geometry.viewport.northeast.lat      0
geometry.viewport.northeast.lng      0
geometry.viewport.southwest.lat      0
geometry.viewport.southwest.lng      0
opening_hours.open_now               5
plus_code.compound_code              0
plus_code.global_code                0
permanently_closed                 194
price_level                        194
PostalCode                           0
Street                               0
city country                         1
dtype: int64


business_status                                  formatted_address  \
0     OPERATIONAL       Löwenstrasse 31-35, 8001 Zürich, Switzerland   
1     OPERATIONAL             Hauptbahnhof, 8001 Zürich, Switzerland   
2     OPERATIONAL  Schaffhauserstrasse 470, 8052 Zürich, Switzerland   
3     OPERATIONAL        Limmatstrasse 152, 8005 Zürich, Switzerland   
4     OPERATIONAL  Birmensdorferstrasse 320, 8055 Zürich, Switzer...   

                                                icon               name  \
0  https://maps.gstatic.com/mapfiles/place_api/ic...             Migros   
1  https://maps.gstatic.com/mapfiles/place_api/ic...  Migros Supermarkt   
2  https://maps.gstatic.com/mapfiles/place_api/ic...  Migros Supermarkt   
3  https://maps.gstatic.com/mapfiles/place_api/ic...  Migros Supermarkt   
4  https://maps.gstatic.com/mapfiles/place_api/ic...  Migros Supermarkt   

                                              photos  \
0  [{'height': 2976, 'html_attributions': ['<a hr...   
1  [{'height': 1960, 'html_attributions': ['<a hr...   
2  [{'height': 4640, 'html_attributions': ['<a hr...   
3  [{'height': 1836, 'html_attributions': ['<a hr...   
4  [{'height': 2204, 'html_attributions': ['<a hr...   

                      place_id  rating                    reference  \
0  ChIJebvE0AUKkEcRli-dLu8vFaM     4.3  ChIJebvE0AUKkEcRli-dLu8vFaM   
1  ChIJS6CT8QgKkEcRwqhtQkVOTUI     4.2  ChIJS6CT8QgKkEcRwqhtQkVOTUI   
2  ChIJzcr5wpsKkEcR8OcT8pamnyI     4.3  ChIJzcr5wpsKkEcR8OcT8pamnyI   
3  ChIJS6CT8QgKkEcRVC558wYywsQ     4.3  ChIJS6CT8QgKkEcRVC558wYywsQ   
4  ChIJlyh8cScKkEcRY3M2wmnjbVI     4.3  ChIJlyh8cScKkEcRY3M2wmnjbVI   

                                               types  user_ratings_total  ...  \
0  [grocery_or_supermarket, supermarket, food, po...                1754  ...   
1  [grocery_or_supermarket, supermarket, food, po...                1246  ...   
2  [grocery_or_supermarket, supermarket, food, po...                 532  ...   
3  [grocery_or_supermarket, supermarket, food, po...                1266  ...   
4  [grocery_or_supermarket, supermarket, food, po...                 414  ...   

   geometry.viewport.southwest.lat  geometry.viewport.southwest.lng  \
0                        47.373931                         8.535093   
1                        47.377074                         8.539151   
2                        47.418227                         8.546886   
3                        47.383983                         8.529818   
4                        47.368534                         8.507371   

   opening_hours.open_now      plus_code.compound_code  plus_code.global_code  \
0                    True  9GGP+4G Zürich, Switzerland            8FVC9GGP+4G   
1                    True  9GHR+84 Zürich, Switzerland            8FVC9GHR+84   
2                    True  CG9X+R8 Zürich, Switzerland            8FVCCG9X+R8   
3                    True  9GPJ+5H Zürich, Switzerland            8FVC9GPJ+5H   
4                    True  9G95+XF Zürich, Switzerland            8FVC9G95+XF   

   permanently_closed price_level PostalCode                    Street  \
0                 NaN         NaN       8001        Löwenstrasse 31-35   
1                 NaN         NaN       8001              Hauptbahnhof   
2                 NaN         NaN       8052   Schaffhauserstrasse 470   
3                 NaN         NaN       8005         Limmatstrasse 152   
4                 NaN         NaN       8055  Birmensdorferstrasse 320   

          city country  
0  Zürich, Switzerland  
1  Zürich, Switzerland  
2  Zürich, Switzerland  
3  Zürich, Switzerland  
4  Zürich, Switzerland  

[5 rows x 24 columns]

In [230]:
df_stores.sort_values(by=['city country']).head(50)

business_status                                  formatted_address  \
7      OPERATIONAL        Albisstrasse 10, 8134 Adliswil, Switzerland   
18     OPERATIONAL      Zürichstrasse 114, 8134 Adliswil, Switzerland   
10     OPERATIONAL  Zürcherstrasse 110/112, 8903 Birmensdorf, Swit...   
13     OPERATIONAL   Zürcherstrasse 15, 8903 Birmensdorf, Switzerland   
18     OPERATIONAL   Ruchwiesenstrasse 2, 8157 Dielsdorf, Switzerland   
11     OPERATIONAL      Neugutstrasse 83, 8600 Dübendorf, Switzerland   
19     OPERATIONAL     Bahnhofstrasse 42, 8703 Erlenbach, Switzerland   
14     OPERATIONAL         Flughofstrasse 3, 8058 Kloten, Switzerland   
9      OPERATIONAL  Schaffhauserstrasse 163 Zürich, 8302 Kloten, S...   
3      OPERATIONAL  Airport Shopping, Airport Shopping, 8302 Klote...   
12     OPERATIONAL      Zürichstrasse 149, 8700 Küsnacht, Switzerland   
10     OPERATIONAL          Zürichstrasse 108, 8123 Maur, Switzerland   
6      OPERATIONAL    Im Zentrum 2, 8102 Oberengstringen, Switzerland   
15     OPERATIONAL    Wallisellerstrasse 8, 8152 Opfikon, Switzerland   
0      OPERATIONAL  Schaffhauserstrasse 29, 8152 Opfikon, Switzerland   
16     OPERATIONAL    Thurgauerstrasse 106, 8152 Opfikon, Switzerland   
7      OPERATIONAL   Adlikerstrasse 295, 8105 Regensdorf, Switzerland   
2      OPERATIONAL         Im Zentrum 1, 8105 Regensdorf, Switzerland   
8      OPERATIONAL      Reitmenstrasse 5, 8952 Schlieren, Switzerland   
9      OPERATIONAL        Güterstrasse 5, 8952 Schlieren, Switzerland   
19     OPERATIONAL      Bahnhofstrasse 2, 8952 Schlieren, Switzerland   
5      OPERATIONAL        Zürcherstrasse 8, 8142 Uitikon, Switzerland   
6      OPERATIONAL  Neue Winterthurerstrasse 2, 8304 Wallisellen, ...   
18     OPERATIONAL  Neue Winterthurerstrasse 99, 8304 Wallisellen,...   
3      OPERATIONAL   Bahnhofstrasse 30, 8304 Wallisellen, Switzerland   
19     OPERATIONAL     Rotfluhstrasse 100, 8702 Zollikon, Switzerland   
17     OPERATIONAL          Dorfstrasse 17, 8126 Zumikon, Switzerland   
5      OPERATIONAL           Seestrasse 339, 8038 Zürich, Switzerland   
6      OPERATIONAL   Pfingstweidstrasse 106, 8005 Zürich, Switzerland   
7      OPERATIONAL    Universitätstrasse 73, 8006 Zürich, Switzerland   
4      OPERATIONAL                Vulkanpl., 8048 Zürich, Switzerland   
8      OPERATIONAL      Halle, Löwenstrasse, 8021 Zürich, Switzerland   
9      OPERATIONAL       Badenerstrasse 333, 8003 Zürich, Switzerland   
10     OPERATIONAL           Hohlstrasse 35, 8004 Zürich, Switzerland   
0      OPERATIONAL       Löwenstrasse 31-35, 8001 Zürich, Switzerland   
19     OPERATIONAL          Vulkanpl. 11-17, 8048 Zürich, Switzerland   
1      OPERATIONAL           Tessinerpl. 10, 8002 Zürich, Switzerland   
15     OPERATIONAL        Badenerstrasse 15, 8004 Zürich, Switzerland   
16     OPERATIONAL       Badenerstrasse 650, 8048 Zürich, Switzerland   
17     OPERATIONAL       Seefeldstrasse 123, 8008 Zürich, Switzerland   
18     OPERATIONAL     Röschibachstrasse 22, 8037 Zürich, Switzerland   
19     OPERATIONAL  Alte Kalchbühlstrasse 15, 8038 Zürich, Switzer...   
0      OPERATIONAL           Löwenstrasse 9, 8001 Zürich, Switzerland   
1      OPERATIONAL      Witikonerstrasse 15, 8032 Zürich, Switzerland   
2      OPERATIONAL     Hofwiesenstrasse 369, 8050 Zürich, Switzerland   
3      OPERATIONAL            Am Wasser 146, 8049 Zürich, Switzerland   
4      OPERATIONAL       Hofwiesenstrasse 5, 8057 Zürich, Switzerland   
5      OPERATIONAL          Hohlstrasse 461, 8048 Zürich, Switzerland   
2      OPERATIONAL        Badenerstrasse 15, 8004 Zürich, Switzerland   
6      OPERATIONAL       Badenerstrasse 812, 8048 Zürich, Switzerland   

                                                 icon  \
7   https://maps.gstatic.com/mapfiles/place_api/ic...   
18  https://maps.gstatic.com/mapfiles/place_api/ic...   
10  https://maps.gstatic.com/mapfiles/place_api/ic...   
13  https://maps.gstatic.com/mapfiles/place_api/ic.

In [231]:
df_stores[['PostalCode']] = df_stores["formatted_address"].str.extract("(\d\d\d\d)",expand=True)
df_stores[['Street','city country']] = df_stores["formatted_address"].str.split(", \d\d\d\d ",expand=True)
df_stores=df_stores.drop(["formatted_address","icon","photos","reference","opening_hours.open_now","plus_code.compound_code","plus_code.global_code","permanently_closed","price_level"],axis=1)

In [235]:
df_stores=df_stores.drop(["geometry.viewport.northeast.lat","geometry.viewport.northeast.lng","geometry.viewport.southwest.lat","geometry.viewport.southwest.lng"],axis=1)
df_stores.head()

business_status               name                     place_id  rating  \
0     OPERATIONAL             Migros  ChIJebvE0AUKkEcRli-dLu8vFaM     4.3   
1     OPERATIONAL  Migros Supermarkt  ChIJS6CT8QgKkEcRwqhtQkVOTUI     4.2   
2     OPERATIONAL  Migros Supermarkt  ChIJzcr5wpsKkEcR8OcT8pamnyI     4.3   
3     OPERATIONAL  Migros Supermarkt  ChIJS6CT8QgKkEcRVC558wYywsQ     4.3   
4     OPERATIONAL  Migros Supermarkt  ChIJlyh8cScKkEcRY3M2wmnjbVI     4.3   

                                               types  user_ratings_total  \
0  [grocery_or_supermarket, supermarket, food, po...                1754   
1  [grocery_or_supermarket, supermarket, food, po...                1246   
2  [grocery_or_supermarket, supermarket, food, po...                 532   
3  [grocery_or_supermarket, supermarket, food, po...                1266   
4  [grocery_or_supermarket, supermarket, food, po...                 414   

   geometry.location.lat  geometry.location.lng PostalCode  \
0              47.375340               8.536317       8001   
1              47.378261               8.540276       8001   
2              47.419532               8.548329       8052   
3              47.385466               8.531396       8005   
4              47.369952               8.508698       8055   

                     Street         city country  
0        Löwenstrasse 31-35  Zürich, Switzerland  
1              Hauptbahnhof  Zürich, Switzerland  
2   Schaffhauserstrasse 470  Zürich, Switzerland  
3         Limmatstrasse 152  Zürich, Switzerland  
4  Birmensdorferstrasse 320  Zürich, Switzerland

In [236]:
df_stores.to_csv("zurich_stores.csv")

In [119]:
stores_near_zurich_MAW = pd.read_csv("DataChallenge/stores_near_zurich_MAW.csv")
stores_near_zurich_GB = pd.read_csv("DataChallenge/stores_near_zurich_GB.csv")

In [120]:
df_merged=pd.concat([stores_near_zurich_MAW,stores_near_zurich_GB],axis=0, ignore_index=True).reset_index(drop=True)
df_merged.sort_values(by="place_id").groupby("geometry.location.lat").head(50)


Unnamed: 0 business_status                                    name  \
347           9     OPERATIONAL   Coop Supermarkt Zürich Schwamendingen   
319           1     OPERATIONAL  Coop Supermarkt Zürich Wipkingerbrücke   
161           5     OPERATIONAL  Coop Supermarkt Zürich Wipkingerbrücke   
396          18     OPERATIONAL  Coop Supermarkt Zürich Wipkingerbrücke   
114          18     OPERATIONAL  Coop Supermarkt Zürich Wipkingerbrücke   
..          ...             ...                                     ...   
281           9     OPERATIONAL                                    Lidl   
29            9     OPERATIONAL                       Migros Supermarkt   
223           8     OPERATIONAL                       Migros Supermarkt   
174          18     OPERATIONAL          Coop Supermarkt Zürich Rütihof   
344           6     OPERATIONAL          Coop Supermarkt Zürich Rütihof   

                        place_id  rating  \
347  ChIJ-0XXwWGgmkcRUk9DeQ9cDqo     3.8   
319  ChIJ-Vog-GkKkEcRSpN3HDvClnY     4.0   
161  ChIJ-Vog-GkKkEcRSpN3HDvClnY     4.0   
396  ChIJ-Vog-GkKkEcRSpN3HDvClnY     4.0   
114  ChIJ-Vog-GkKkEcRSpN3HDvClnY     4.0   
..                           ...     ...   
281  ChIJzwIBDXmhmkcR26Upx4vry2A     4.2   
29   ChIJzwvLd7ugmkcR0PWkijPdm54     3.9   
223  ChIJzwvLd7ugmkcR0PWkijPdm54     3.9   
174  ChIJzxVi73QLkEcRqPs8Nd-0uWU     3.7   
344  ChIJzxVi73QLkEcRqPs8Nd-0uWU     3.7   

                                                 types  user_ratings_total  \
347  ['supermarket', 'grocery_or_supermarket', 'foo...                 454   
319  ['supermarket', 'grocery_or_supermarket', 'foo...                 254   
161  ['supermarket', 'grocery_or_supermarket', 'foo...                 254   
396  ['supermarket', 'grocery_or_supermarket', 'foo...                 254   
114  ['supermarket', 'grocery_or_supermarket', 'foo...                 254   
..                                                 ...                 ...   
281  ['supermarket', 'convenience_store', 'grocery_...                 503   
29   ['grocery_or_supermarket', 'supermarket', 'foo...                  99   
223  ['grocery_or_supermarket', 'supermarket', 'foo...                  99   
174  ['supermarket', 'grocery_or_supermarket', 'foo...                  86   
344  ['supermarket', 'grocery_or_supermarket', 'foo...                  86   

     geometry.location.lat  geometry.location.lng  PostalCode  ...  \
347              47.404916               8.571550        8051  ...   
319              47.392558               8.524622        8037  ...   
161              47.392558               8.524622        8037  ...   
396              47.392558               8.524622        8037  ...   
114              47.392558               8.524622        8037  ...   
..                     ...                    ...         ...  ...   
281              47.395218               8.638227        8600  ...   
29               47.379304               8.559396        8044  ...   
223              47.379304               8.559396        8044  ...   
174              47.413741               8.479399        8049  ...   
344              47.413741               8.479399        8049  ...   

                       reference geometry.viewport.northeast.lat  \
347  ChIJ-0XXwWGgmkcRUk9DeQ9cDqo                       47.406209   
319  ChIJ-Vog-GkKkEcRSpN3HDvClnY                       47.393932   
161                          NaN                             NaN   
396  ChIJ-Vog-GkKkEcRSpN3HDvClnY                       47.393932   
114                          NaN                             NaN   
..                           ...                             ...   
281  ChIJzwIBDXmhmkcR26Upx4vry2A                       47.396499   
29                           NaN                             NaN   
223  ChIJzwvLd7ugmkcR0PWkijPdm54                       47.380610   
174                          NaN                             NaN   
344  ChIJzxVi73QLkEcRqPs8Nd-0uWU                 

In [121]:
duplicateRowsDF = df_merged[df_merged.duplicated()]
print("Duplicate Rows except first occurrence based on all columns are :")
print(duplicateRowsDF)
print(list(df_merged.columns))
df_merged=df_merged.drop(columns=["Unnamed: 0"]).sort_values(by=['geometry.location.lat'])

df_merged[df_merged['types'].apply(lambda x :  ('supermarket' not in x))]
df_merged=df_merged.drop(['geometry.viewport.northeast.lat','geometry.viewport.northeast.lng','geometry.viewport.southwest.lat','geometry.viewport.southwest.lng',"photos","icon","formatted_address","reference","opening_hours.open_now","plus_code.compound_code","plus_code.global_code","permanently_closed","price_level"],axis=1)
df_merged = df_merged.drop_duplicates(subset=['place_id'], keep="first") 
df_merged=df_merged.reset_index()
df_merged=df_merged.drop(["index"],axis=1)
df_merged.head()
df_merged.to_csv("stores_near_zurich_merged_duplicates_removed.csv")
#df_merged.shape

Duplicate Rows except first occurrence based on all columns are :
     Unnamed: 0 business_status                            name  \
153          17     OPERATIONAL  Coop Supermarkt Zürich Seefeld   

                        place_id  rating  \
153  ChIJdUL9H02nmkcR9sm4Ipv4OMU     4.1   

                                                 types  user_ratings_total  \
153  ['supermarket', 'grocery_or_supermarket', 'foo...                 343   

     geometry.location.lat  geometry.location.lng  PostalCode  ... reference  \
153              47.357934               8.554554        8008  ...       NaN   

    geometry.viewport.northeast.lat geometry.viewport.northeast.lng  \
153                             NaN                             NaN   

    geometry.viewport.southwest.lat geometry.viewport.southwest.lng  \
153                             NaN                             NaN   

    opening_hours.open_now  plus_code.compound_code  plus_code.global_code  \
153                    NaN  

In [5]:
df_merged[df_merged['types'].apply(lambda x :  ('supermarket' not in x))]
df_merged=df_merged.drop(['geometry.viewport.northeast.lat','geometry.viewport.northeast.lng','geometry.viewport.southwest.lat','geometry.viewport.southwest.lng',"photos","icon","formatted_address","reference","opening_hours.open_now","plus_code.compound_code","plus_code.global_code","permanently_closed","price_level"],axis=1)
df_merged = df_merged.drop_duplicates(subset=['place_id'], keep="first") 
df_merged.head()
df_merged.to_csv("stores_near_zurich_merged_duplicates_removed.csv")
df_merged.shape


NameError: name 'df_merged' is not defined

In [68]:
df_shops=pd.read_csv("stores_near_zurich_merged_duplicates_removed.csv")
df_quarter_middle=pd.read_excel("Data/Zurich_Quarters.xlsx")
df_quarter_middle.head(50)
#checked successcully that umlauts do work by by putting them into string
df_quarter_middle[df_quarter_middle["Name"]=="Höngg"]
print("shape 1 before, ",df_shops.shape)
df_merged1=df_shops.merge(right=df_quarter_middle,how="left",left_on='PostalCode', right_on='PLZ')
df_merged1.to_csv("test.csv")

shape 1 before,  (205, 12)


Merge attempt from above failed, in that sense that PostalCode and quarters are in fact not distinct. There are three districts with each two postal codes, but other districts share the postal code. Idea is not to merge the both Zurich quarter related overview tables, order these by postal code, and then merge the information.  

It seems that on any string in Statistische_Quartiere.xlsx the end includes a whitespace. This leads to a huge failure in the automated merging

In [81]:
df_quarter_1=pd.read_excel("Data/Zurich_Quarters.xlsx")
df_quarter_2=pd.read_excel("DataChallenge/Statistische_Quartiere.xlsx")
df_quarter_2["Statistische Quartiere "]=df_quarter_2["Statistische Quartiere "].str.strip(" ")
print(list(df_quarter_2.columns))
df_quarters=df_quarter_1.merge(right=df_quarter_2,how="left",left_on='Name', right_on='Statistische Quartiere ')
df_quarters.sort_values(by="Stadtkreis ").head()
df_quarters=df_quarters.drop(['Statistische Quartiere '],axis=1)
df_quarters.sort_values(by="Stadtkreis ").head(40)
#df_quarters.dtypes

['Stadtkreis ', 'Statistische Quartiere ', 'BFS-Code ', 'Eingemeindung ', 'Fläche\nin km² ', 'Einwohner \nDez. 2018', 'Einwohner\nDez. 2013 ', 'Einwohner\nDez. 2005 ', 'Ausländer \nDez. 2018']


PLZ                  Name       Typ  QuarSor Stadtkreis   BFS-Code   \
26  8001               Rathaus  Quartier       11     Kreis 1     261011   
6   8001                  City  Quartier       14     Kreis 1     261014   
21  8001             Lindenhof  Quartier       13     Kreis 1     261013   
16  8001           Hochschulen  Quartier       12     Kreis 1     261012   
36  8037             Wipkingen  Quartier      102   Kreis 10      261102   
17  8049                 Höngg  Quartier      101   Kreis 10      261101   
0   8046             Affoltern  Quartier      111   Kreis 11      261111   
29  8052               Seebach  Quartier      119   Kreis 11      261119   
25  8057              Oerlikon  Quartier      115   Kreis 11      261115   
24  8050              Oerlikon  Quartier      115   Kreis 11      261115   
28  8051  Schwamendingen-Mitte  Quartier      122    Kreis 12     261122   
27  8050               Saatlen  Quartier      121    Kreis 12     261121   
15  8051            Hirzenbach  Quartier      123    Kreis 12     261123   
20  8041              Leimbach  Quartier       23    Kreis 2      261023   
38  8038           Wollishofen  Quartier       21    Kreis 2      261021   
7   8002                  Enge  Quartier       24    Kreis 2      261024   
11  8055           Friesenberg  Quartier       33     Kreis 3     261033   
31  8003              Sihlfeld  Quartier       34     Kreis 3     261034   
2   8003          Alt-Wiedikon  Quartier       31     Kreis 3     261031   
10  8045           Friesenberg  Quartier       33     Kreis 3     261033   
3   8045          Alt-Wiedikon  Quartier       31     Kreis 3     261031   
13  8004                  Hard  Quartier       44     Kreis 4     261044   
35  8004                  Werd  Quartier       41     Kreis 4     261041   
19  8004           Langstrasse  Quartier       42     Kreis 4     261042   
8   8005           Escher Wyss  Quartier       52     Kreis 5     261052   
12  8005         Gewerbeschule  Quartier       51     Kreis 5     261051   
32  8006           Unterstrass  Quartier       61    Kreis 6      261061   
33  8057           Unterstrass  Quartier       61    Kreis 6      261061   
23  8006            Oberstrass  Quartier       63    Kreis 6      261063   
14  8032            Hirslanden  Quartier       73    Kreis 7      261073   
9   8044              Fluntern  Quartier       71    Kreis 7      261071   
37  8053               Witikon  Quartier       74    Kreis 7      261074   
18  8032             Hottingen  Quartier       72    Kreis 7      261072   
22  8008             Mühlebach  Quartier       82     Kreis 8     261082   
30  8008               Seefeld  Quartier       81     Kreis 8     261081   
34  8008               Weinegg  Quartier       83     Kreis 8     261083   
4   8048            Altstetten  Quartier       92    Kreis 9      261092   
1   8047           Albisrieden  Quartier       91    Kreis 9      261091   
5   8064            Altstetten  Quartier       92    Kreis 9      261092   

   Eingemeindung   Fläche\nin km²   Einwohner \nDez. 2018  \
26      vor 1893              0.38                   3267   
6       vor 1893              0.64                    829   
21      vor 1893              0.23                    990   
16      vor 1893              0.56                    664   
36           1893             2.11                  16321   
17           1934             6.98                  24020   
0            1934             6.04                  26562   
29           1934             4.72                  25568   
25           1934             2.67                  23214   
24           1934             2.67                  23214   
28           1934             2.23                  11100   
27           1934             1.13                   8582   
15           1934             2.62                  12801   
20           1893             2.92                   6320   
38           1893             5.75               

If we consider postal codes, Kreis 1, Kreis 4, Kreis 5, and Kreis 8 need to be merged. In Google Maps API we have info from PostalCodes but not from Kreise yet.

In [82]:
df_quarters['Ausländer \nDez. 2018'] = df_quarters["Ausländer \nDez. 2018"].str.replace(',', '.', regex=True).str.strip(" % ").astype('float64')
print("after",df_quarters['Ausländer \nDez. 2018'][0])

after 33.0


In [83]:
df_quarters['Ausländer \nDez. 2018']=round(0.01*df_quarters['Ausländer \nDez. 2018']*df_quarters['Einwohner \nDez. 2018'],0)
df_quarters['Ausländer \nDez. 2018']=df_quarters['Ausländer \nDez. 2018'].astype('int64')
print(df_quarters.dtypes,"after",df_quarters['Ausländer \nDez. 2018'][0])

PLZ                        int64
Name                      object
Typ                       object
QuarSor                    int64
Stadtkreis                object
BFS-Code                   int64
Eingemeindung             object
Fläche\nin km²           float64
Einwohner \nDez. 2018      int64
Einwohner\nDez. 2013       int64
Einwohner\nDez. 2005       int64
Ausländer \nDez. 2018      int64
dtype: object after 8765


In [84]:
df_quarters.sort_values(by="PLZ").head(40)

PLZ                  Name       Typ  QuarSor Stadtkreis   BFS-Code   \
21  8001             Lindenhof  Quartier       13     Kreis 1     261013   
16  8001           Hochschulen  Quartier       12     Kreis 1     261012   
26  8001               Rathaus  Quartier       11     Kreis 1     261011   
6   8001                  City  Quartier       14     Kreis 1     261014   
7   8002                  Enge  Quartier       24    Kreis 2      261024   
31  8003              Sihlfeld  Quartier       34     Kreis 3     261034   
2   8003          Alt-Wiedikon  Quartier       31     Kreis 3     261031   
35  8004                  Werd  Quartier       41     Kreis 4     261041   
13  8004                  Hard  Quartier       44     Kreis 4     261044   
19  8004           Langstrasse  Quartier       42     Kreis 4     261042   
8   8005           Escher Wyss  Quartier       52     Kreis 5     261052   
12  8005         Gewerbeschule  Quartier       51     Kreis 5     261051   
23  8006            Oberstrass  Quartier       63    Kreis 6      261063   
32  8006           Unterstrass  Quartier       61    Kreis 6      261061   
34  8008               Weinegg  Quartier       83     Kreis 8     261083   
30  8008               Seefeld  Quartier       81     Kreis 8     261081   
22  8008             Mühlebach  Quartier       82     Kreis 8     261082   
18  8032             Hottingen  Quartier       72    Kreis 7      261072   
14  8032            Hirslanden  Quartier       73    Kreis 7      261073   
36  8037             Wipkingen  Quartier      102   Kreis 10      261102   
38  8038           Wollishofen  Quartier       21    Kreis 2      261021   
20  8041              Leimbach  Quartier       23    Kreis 2      261023   
9   8044              Fluntern  Quartier       71    Kreis 7      261071   
3   8045          Alt-Wiedikon  Quartier       31     Kreis 3     261031   
10  8045           Friesenberg  Quartier       33     Kreis 3     261033   
0   8046             Affoltern  Quartier      111   Kreis 11      261111   
1   8047           Albisrieden  Quartier       91    Kreis 9      261091   
4   8048            Altstetten  Quartier       92    Kreis 9      261092   
17  8049                 Höngg  Quartier      101   Kreis 10      261101   
24  8050              Oerlikon  Quartier      115   Kreis 11      261115   
27  8050               Saatlen  Quartier      121    Kreis 12     261121   
15  8051            Hirzenbach  Quartier      123    Kreis 12     261123   
28  8051  Schwamendingen-Mitte  Quartier      122    Kreis 12     261122   
29  8052               Seebach  Quartier      119   Kreis 11      261119   
37  8053               Witikon  Quartier       74    Kreis 7      261074   
11  8055           Friesenberg  Quartier       33     Kreis 3     261033   
25  8057              Oerlikon  Quartier      115   Kreis 11      261115   
33  8057           Unterstrass  Quartier       61    Kreis 6      261061   
5   8064            Altstetten  Quartier       92    Kreis 9      261092   

   Eingemeindung   Fläche\nin km²   Einwohner \nDez. 2018  \
21      vor 1893              0.23                    990   
16      vor 1893              0.56                    664   
26      vor 1893              0.38                   3267   
6       vor 1893              0.64                    829   
7            1893             2.40                   9634   
31           1893             1.64                  21680   
2            1893             1.85                  17956   
35           1893             0.31                   4455   
13           1893             1.46                  13163   
19           1893             1.13                  11111   
8            1893             1.27                   6066   
12           1893             0.73                   9513   
23           1893             2.64                  10927   
32           1893             2.46                  23394   
34           1893             1.72               

In [121]:
df_kreis_pop=pd.read_excel("DataChallenge/Statistische_Quartiere.xlsx")
df_kreis_pop['Ausländer \nDez. 2018'] = df_kreis_pop["Ausländer \nDez. 2018"].str.replace(',', '.', regex=True).str.strip(" % ").astype('float64')
df_kreis_pop['Ausländer \nDez. 2018']=round(0.01*df_kreis_pop['Ausländer \nDez. 2018']*df_kreis_pop['Einwohner \nDez. 2018'],0)
df_kreis_pop['Ausländer \nDez. 2018']=df_kreis_pop['Ausländer \nDez. 2018'].astype('int64')
#list(df_kreis_pop.columns)
df_kreis_pop=df_kreis_pop.drop(columns=["Eingemeindung ","BFS-Code "]).groupby("Stadtkreis ").agg("sum").reset_index()
#df_kreis_pop.head(100)
df_kreis_pop.to_csv("kreis_population.csv")
df_kreis_pop.head(100)

Stadtkreis   Fläche\nin km²   Einwohner \nDez. 2018  Einwohner\nDez. 2013   \
0      Kreis 1             1.81                   5750                   5565   
1    Kreis 10              9.09                  40341                  37416   
2    Kreis 11             13.43                  75344                  70748   
3     Kreis 12             5.98                  32483                  29642   
4     Kreis 2             11.07                  34877                  30503   
5      Kreis 3             8.64                  50569                  48333   
6      Kreis 4             2.90                  28729                  27589   
7      Kreis 5             2.00                  15579                  13636   
8     Kreis 6              5.10                  34321                  32328   
9     Kreis 7             15.02                  38191                  36116   
10     Kreis 8             4.80                  16788                  15912   
11    Kreis 9             12.07                  55765                  50261   

    Einwohner\nDez. 2005   Ausländer \nDez. 2018  
0                    5572                   1758  
1                   36219                  10632  
2                   58197                  27704  
3                   28314                  11638  
4                   28834                  11167  
5                   45885                  15014  
6                   27487                  10613  
7                   12417                   5020  
8                   29415                   9771  
9                   34148                  10846  
10                  15343                   6052  
11                  45504                  18056

OK seems a split into quarters gets us only part of the story, next idea was to look at Kreis, but then 8050 is e.g. split between two Kreise: 11 and 12. Thus even Kreis merge does not give us a one to one correspondence.

In [6]:
#8001,8008
#now group df into postal code df to merge it later 
df_to_study=df_quarters.drop(["QuarSor","BFS-Code "],axis=1)
df_plz=df_quarters.drop(["QuarSor","BFS-Code "],axis=1).groupby("PLZ").agg("sum")
df_plz.head(40)

Fläche\nin km²   Einwohner \nDez. 2018  Einwohner\nDez. 2013   \
PLZ                                                                   
8001             1.81                   5750                   5565   
8002             2.40                   9634                   8836   
8003             3.49                  39636                  37637   
8004             2.90                  28729                  27589   
8005             2.00                  15579                  13636   
8006             5.10                  34321                  32328   
8008             4.80                  16788                  15912   
8032             7.25                  18753                  17993   
8037             2.11                  16321                  15835   
8038             5.75                  18923                  15937   
8041             2.92                   6320                   5730   
8044             2.84                   8485                   7856   
8045             7.00                  28889                  27402   
8046             6.04                  26562                  25082   
8047             4.60                  22304                  19146   
8048             7.47                  33461                  31115   
8049             6.98                  24020                  21581   
8050             3.80                  31796                  28938   
8051             4.85                  23901                  22362   
8052             4.72                  25568                  24008   
8053             4.93                  10953                  10267   
8055             5.15                  10933                  10696   
8057             5.13                  46608                  43534   
8064             7.47                  33461                  31115   

      Einwohner\nDez. 2005   Ausländer \nDez. 2018  
PLZ                                                 
8001                   5572                   1758  
8002                   8375                   3536  
8003                  35525                  13013  
8004                  27487                  10613  
8005                  12417                   5020  
8006                  29415                   9771  
8008                  15343                   6052  
8032                  16959                   5393  
8037                  15446                   4603  
8038                  15592                   5507  
8041                   4867                   2124  
8044                   7325                   2605  
8045                  25331                   8250  
8046                  18733                   8765  
8047                  17226                   6044  
8048                  28278                  12012  
8049                  20773                   6029  
8050                  26312                  11363  
8051                  21587                   9166  
8052                  19879                  10048  
8053                   9864                   2848  
8055                  10360                   2001  
8057                  39506                  15395  
8064                  28278                  12012

Issue with THIS approach: in case two postal codes are distributed to even two Kreises (who came up with that), we "hide" this effect, and now think about the complication of Oerlikon being split between two postal codes, and then Oerlikon a) gets part of Saatlen from a different Kreis, oh yes who would have guessed that would happen.

In [87]:
df_shops_plz=df_shops.merge(right=df_plz,how="left",left_on='PostalCode', right_on='PLZ')
print(df_shops_plz.shape)
df_shops_plz.head(40)


(205, 17)


Unnamed: 0 business_status                                        name  \
0            0     OPERATIONAL                           Migros Supermarkt   
1            1     OPERATIONAL                           Migros Supermarkt   
2            2     OPERATIONAL                           Migros Supermarkt   
3            3     OPERATIONAL             Coop Supermarkt Zürich Leimbach   
4            4     OPERATIONAL             Coop Pronto Shop mit Tankstelle   
5            5     OPERATIONAL                                         VOI   
6            6     OPERATIONAL                           Migros Supermarkt   
7            7     OPERATIONAL                           Migros Supermarkt   
8            8     OPERATIONAL                           Migros Supermarkt   
9            9     OPERATIONAL          Coop Supermarkt Zürich Wollishofen   
10          10     OPERATIONAL                              migrolino Shop   
11          11     OPERATIONAL                                 ALDI Zürich   
12          12     OPERATIONAL                           Migros Supermarkt   
13          13     OPERATIONAL                           Migros Florissimo   
14          14     OPERATIONAL  Coop Supermarkt Zürich Wollishofen Seestr.   
15          15     OPERATIONAL                           Migros Supermarkt   
16          16     OPERATIONAL                              migrolino Shop   
17          17     OPERATIONAL        Coop Supermarkt Zürich Tiefenbrunnen   
18          18     OPERATIONAL                           Migros Supermarkt   
19          19     OPERATIONAL             Coop Zürich Sihlcity Supermarkt   
20          20     OPERATIONAL                                   Coop City   
21          21     OPERATIONAL                           Migros Supermarkt   
22          22     OPERATIONAL     Coop Vitality Apotheke Zürich Sihl City   
23          23     OPERATIONAL              Coop Supermarkt Zürich Seefeld   
24          24     OPERATIONAL                           Migros Supermarkt   
25          25     OPERATIONAL                           Migros Supermarkt   
26          26     OPERATIONAL                                 ALDI Zürich   
27          27     OPERATIONAL                           Migros Supermarkt   
28          28     OPERATIONAL             Coop Pronto Shop mit Tankstelle   
29          29     OPERATIONAL                              migrolino Shop   
30          30     OPERATIONAL     Coop Supermarkt Zürich Witikonerstrasse   
31          31     OPERATIONAL  Coop Supermarkt Zürich Zentrum Friesenberg   
32          32     OPERATIONAL                                Lidl Schweiz   
33          33     OPERATIONAL            Coop Supermarkt Zürich Brotgasse   
34          34     OPERATIONAL                            Coop Pronto Shop   
35          35     OPERATIONAL                           Migros Supermarkt   
36          36     OPERATIONAL           Coop Supermarkt Zürich Kreuzplatz   
37          37     OPERATIONAL                            Coop Pronto Shop   
38          38     OPERATIONAL         Coop Supermarkt Zürich Enge Bahnhof   
39          39     OPERATIONAL                           Migros Supermarkt   

                       place_id  rating  \
0   ChIJR2S8OUGmmkcR8rApXQaV9Ck     4.1   
1   ChIJz36CXYqnmkcRPGj6agSMWlo     4.3   
2   ChIJa1QxRlGmmkcRkAjn_uA_XJc     4.3   
3   ChIJTUhiXjIIkEcRJo4d0p5UxZg     4.2   
4   ChIJvRhhF4CnmkcRfikKLNDOMwo     4.0   
5   ChIJ94CtOdYJkEcRRDxXKaxpKDI     4.0   
6   ChIJ6cfaKs4JkEcRgtb1q4szNRU     4.1   
7   ChIJ1dktgpmmmkcRHICUq60HyXE     4.3   
8   ChIJwcVoQRynmkcRM-mEXgEmq14     4.3   
9   ChIJtd5NWdgJkEcR2n9icvFvjqI     4.3   
10  ChIJYzNJO8UJkEcRPiyA9AoazPA     3.8   
11  ChIJgftGuzAJkEcRnHASrlZ4gYU     4.3   
12  ChIJA0Pf5N4JkEcRD2eKbftCaxc     4.3   
13  ChIJz25eCd4JkEcROqCsuPWZ35A     4.6   
14  ChIJHd1I598JkEcRelTmD3KV3b8     4.0   
15  ChIJEVskDgWkmkcR_tbIQDJOw7A     4.1   
16  ChIJ4zS5ckenmkcRgG7e3h-NwcQ     4.4   
17  ChIJcczfo0enmkcRcVGnSsAbTPE

In [89]:
df_shops_plz.to_csv("stores_near_zurich_merged_duplicates_removed_plzinfo_V0.csv")

PostalCodes are differently distributed than Quarters. Two Quarters can have the same postal code, within one quarter there can be two postal codes. What I decided to do is the following: 
- Consider 8003 as relevant Postal Code for Alt-Wiedikon, then Alt-Wiedikon will be merged with Sihlfeld later on for 8003 area and population, drop 8045 from Alt-Wiedikon, as well as 8045 from Friesenberg, replace 8045 with 8003
- Consider 8055 as Postal Code from Friesenberg, drop 8045 (like for Alt-Wiedikon), this one goes to 8003 for 8045
- Consider 8006 as value for Unterstrass, drop 8057 from Unterstrass, merge Unterstrass with Oberstrass, replace all 8057 with 8006
- Consider 8050 for Oerlikon, drop 8057 from Oerlikon, merge 8050 with Saatlen, 8050 agrees most with Kreis 11
- Drop 8064 for Alstetten, consider 8048 for all Altstetten

In [13]:
df_shopsV1=pd.read_csv("stores_near_zurich_merged_duplicates_removed.csv")
df_shopsV1['PostalCode'] = df_shopsV1['PostalCode'].replace([8045,8057,8064],[8003,8006,8048])
df_shopsV1.to_csv('testV1.csv')

In [14]:
df_shops_plz_V1=df_shopsV1.merge(right=df_plz,how="left",left_on='PostalCode', right_on='PLZ')
print(df_shops_plz_V1.shape)
df_shops_plz_V1.reset_index()
df_shops_plz_V1=df_shops_plz_V1.drop(["Unnamed: 0"],axis=1)
df_shops_plz_V1.head(40)
df_shops_plz_V1.to_csv("stores_near_zurich_merged_duplicates_removed_plzinfo_V1.csv")

(205, 17)


In [69]:
df_plz2=df_quarters.drop(["QuarSor","BFS-Code "],axis=1).groupby(["PLZ","Stadtkreis "]).agg("sum").reset_index()
df_plz2.head(40)

PLZ Stadtkreis   Fläche\nin km²   Einwohner \nDez. 2018  \
0   8001     Kreis 1             1.81                   5750   
1   8002    Kreis 2              2.40                   9634   
2   8003     Kreis 3             3.49                  39636   
3   8004     Kreis 4             2.90                  28729   
4   8005     Kreis 5             2.00                  15579   
5   8006    Kreis 6              5.10                  34321   
6   8008     Kreis 8             4.80                  16788   
7   8032    Kreis 7              7.25                  18753   
8   8037   Kreis 10              2.11                  16321   
9   8038    Kreis 2              5.75                  18923   
10  8041    Kreis 2              2.92                   6320   
11  8044    Kreis 7              2.84                   8485   
12  8045     Kreis 3             7.00                  28889   
13  8046   Kreis 11              6.04                  26562   
14  8047    Kreis 9              4.60                  22304   
15  8048    Kreis 9              7.47                  33461   
16  8049   Kreis 10              6.98                  24020   
17  8050   Kreis 11              2.67                  23214   
18  8050    Kreis 12             1.13                   8582   
19  8051    Kreis 12             4.85                  23901   
20  8052   Kreis 11              4.72                  25568   
21  8053    Kreis 7              4.93                  10953   
22  8055     Kreis 3             5.15                  10933   
23  8057   Kreis 11              2.67                  23214   
24  8057    Kreis 6              2.46                  23394   
25  8064    Kreis 9              7.47                  33461   

    Einwohner\nDez. 2013   Einwohner\nDez. 2005   
0                    5565                   5572  
1                    8836                   8375  
2                   37637                  35525  
3                   27589                  27487  
4                   13636                  12417  
5                   32328                  29415  
6                   15912                  15343  
7                   17993                  16959  
8                   15835                  15446  
9                   15937                  15592  
10                   5730                   4867  
11                   7856                   7325  
12                  27402                  25331  
13                  25082                  18733  
14                  19146                  17226  
15                  31115                  28278  
16                  21581                  20773  
17                  21658                  19585  
18                   7280                   6727  
19                  22362                  21587  
20                  24008                  19879  
21                  10267                   9864  
22                  10696                  10360  
23                  21658                  19585  
24                  21876                  19921  
25                  31115                  28278

In [72]:
df_plz2=df_quarters.drop(["QuarSor","BFS-Code ","PLZ"],axis=1).groupby(["Stadtkreis "]).agg("sum").reset_index()

pandas.core.frame.DataFrame

In [59]:
#list(df_plz2.columns)
df_kreis_plz=df_plz2.loc[:,("PLZ","Stadtkreis ")]
df_kreis_plz.head(20)
df_kreis_plz=df_kreis_plz.drop(df_kreis_plz.loc[(df_kreis_plz["PLZ"]==8050) & (df_kreis_plz["Stadtkreis "]=="Kreis 12")].index)
#aggregate information by Kreis
df_kreis_plz["KreisNumber"]=df_kreis_plz["Stadtkreis "].str.strip("Kreis ")
df_kreis_plz.head(20)
df_kreis_plz.to_csv("PLZ_Stadtkreis_Mapping.csv")

In [134]:
df_kreis_plz_bef=pd.read_csv("PLZ_Stadtkreis_Mapping.csv")
df_shops_plz_bef=pd.read_csv("stores_near_zurich_merged_duplicates_removed_plzinfo_V1.csv")
print(df_shops_plz_bef.shape)
df_shops_aft=df_shops_plz_bef.merge(right=df_kreis_plz_bef,how="left",left_on='PostalCode', right_on='PLZ')
df_shops_aft=df_shops_aft.drop(columns=["Unnamed: 0_x","Unnamed: 0_y","PLZ"])
df_shops_aft=df_shops_aft.rename(columns=lambda x: x.strip())
df_shops_aft.head(30)
#print(list(df_shops_aft.columns))
#df_shops_aft.to_csv("stores_near_zurich_merged_duplicates_removed_plzinfo_V2.csv")

(205, 17)


business_status                                        name  \
0      OPERATIONAL                           Migros Supermarkt   
1      OPERATIONAL                           Migros Supermarkt   
2      OPERATIONAL                           Migros Supermarkt   
3      OPERATIONAL             Coop Supermarkt Zürich Leimbach   
4      OPERATIONAL             Coop Pronto Shop mit Tankstelle   
5      OPERATIONAL                                         VOI   
6      OPERATIONAL                           Migros Supermarkt   
7      OPERATIONAL                           Migros Supermarkt   
8      OPERATIONAL                           Migros Supermarkt   
9      OPERATIONAL          Coop Supermarkt Zürich Wollishofen   
10     OPERATIONAL                              migrolino Shop   
11     OPERATIONAL                                 ALDI Zürich   
12     OPERATIONAL                           Migros Supermarkt   
13     OPERATIONAL                           Migros Florissimo   
14     OPERATIONAL  Coop Supermarkt Zürich Wollishofen Seestr.   
15     OPERATIONAL                           Migros Supermarkt   
16     OPERATIONAL                              migrolino Shop   
17     OPERATIONAL        Coop Supermarkt Zürich Tiefenbrunnen   
18     OPERATIONAL                           Migros Supermarkt   
19     OPERATIONAL             Coop Zürich Sihlcity Supermarkt   
20     OPERATIONAL                                   Coop City   
21     OPERATIONAL                           Migros Supermarkt   
22     OPERATIONAL     Coop Vitality Apotheke Zürich Sihl City   
23     OPERATIONAL              Coop Supermarkt Zürich Seefeld   
24     OPERATIONAL                           Migros Supermarkt   
25     OPERATIONAL                           Migros Supermarkt   
26     OPERATIONAL                                 ALDI Zürich   
27     OPERATIONAL                           Migros Supermarkt   
28     OPERATIONAL             Coop Pronto Shop mit Tankstelle   
29     OPERATIONAL                              migrolino Shop   

                       place_id  rating  \
0   ChIJR2S8OUGmmkcR8rApXQaV9Ck     4.1   
1   ChIJz36CXYqnmkcRPGj6agSMWlo     4.3   
2   ChIJa1QxRlGmmkcRkAjn_uA_XJc     4.3   
3   ChIJTUhiXjIIkEcRJo4d0p5UxZg     4.2   
4   ChIJvRhhF4CnmkcRfikKLNDOMwo     4.0   
5   ChIJ94CtOdYJkEcRRDxXKaxpKDI     4.0   
6   ChIJ6cfaKs4JkEcRgtb1q4szNRU     4.1   
7   ChIJ1dktgpmmmkcRHICUq60HyXE     4.3   
8   ChIJwcVoQRynmkcRM-mEXgEmq14     4.3   
9   ChIJtd5NWdgJkEcR2n9icvFvjqI     4.3   
10  ChIJYzNJO8UJkEcRPiyA9AoazPA     3.8   
11  ChIJgftGuzAJkEcRnHASrlZ4gYU     4.3   
12  ChIJA0Pf5N4JkEcRD2eKbftCaxc     4.3   
13  ChIJz25eCd4JkEcROqCsuPWZ35A     4.6   
14  ChIJHd1I598JkEcRelTmD3KV3b8     4.0   
15  ChIJEVskDgWkmkcR_tbIQDJOw7A     4.1   
16  ChIJ4zS5ckenmkcRgG7e3h-NwcQ     4.4   
17  ChIJcczfo0enmkcRcVGnSsAbTPE     3.9   
18  ChIJI8HnhsoOkEcRtrizPct-sG0     4.3   
19  ChIJFwx7xe8JkEcRDrrE0f9X6mg     4.4   
20  ChIJp3j71fEJkEcRBzbWCpR8iMg     4.3   
21  ChIJvSPwpTGnmkcRfQMgzGhOoYI     4.1   
22  ChIJWzO81-8JkEcREO-U8fsZT-4     3.9   
23  ChIJdUL9H02nmkcR9sm4Ipv4OMU     4.1   
24  ChIJj0lFi-0JkEcRhd1Rz4cxhXc     4.3   
25  ChIJs2st406nmkcRAbmB1ZVvu5Y     4.2   
26  ChIJxXEJCJmnmkcR0vOQ6iIwg8g     4.3   
27  ChIJVbosGiqnmkcRaO1Fr_17AEQ     4.3   
28  ChIJu-tIjUUJkEcRW4pEeesOZj8     4.2   
29  ChIJm5NFmvoJkEcR09JW1-3sOxc     4.2   

                                                types  user_ratings_total  \
0   ['grocery_or_supermarket', 'supermarket', 'foo...                 233   
1   ['grocery_or_supermarket', 'supermarket', 'foo...                 660   
2   ['grocery_or_supermarket', 'supermarket', 'foo...                 117   
3   ['supermarket', 'grocery_or_supermarket', 'foo...                 140   
4   ['gas_station', 'supermarket', 'grocery_or_sup...                 306   
5   ['supermarket', 'bakery', 'grocery_or_supermar...                 125   
6   ['grocery_or_supermarket', 'supermarket', 'foo...                 175   
7   ['

Population per Kreis determined from statistics file previously. Now add this information into stores file as well, the Kreis to PostalCode assignment has been done previously. 

In [146]:
df_shops_bef=pd.read_csv("stores_near_zurich_merged_duplicates_removed_plzinfo_V2.csv")
df_kreis_pop=pd.read_csv("kreis_population.csv")
print(list(df_shops_bef.columns))
print(list(df_kreis_pop.columns))
df_kreis_pop=df_kreis_pop.rename(columns=lambda x: x.strip(" "))
df_shops_bef=df_shops_bef.rename(columns=lambda x: x.strip(" "))
df_shops_aft=df_shops_bef.drop(columns=['Unnamed: 0']).merge(right=df_kreis_pop.drop(columns=['Unnamed: 0']),
                                                             how="left",left_on='Stadtkreis', right_on='Stadtkreis',
                                                             suffixes=('_PLZ', '_Kreis'))
##df_shops_aft.head()
print(list(df_shops_aft.columns))
df_shops_aft.to_csv("stores_near_zurich_merged_duplicates_removed_plz_kreis_info_FINAL.csv")

['Unnamed: 0', 'business_status', 'name', 'place_id', 'rating', 'types', 'user_ratings_total', 'geometry.location.lat', 'geometry.location.lng', 'PostalCode', 'Street', 'city country', 'Fläche\nin km² ', 'Einwohner \nDez. 2018', 'Einwohner\nDez. 2013 ', 'Einwohner\nDez. 2005 ', 'Ausländer \nDez. 2018', 'Stadtkreis ', 'KreisNumber']
['Unnamed: 0', 'Stadtkreis ', 'Fläche\nin km² ', 'Einwohner \nDez. 2018', 'Einwohner\nDez. 2013 ', 'Einwohner\nDez. 2005 ', 'Ausländer \nDez. 2018']
['business_status', 'name', 'place_id', 'rating', 'types', 'user_ratings_total', 'geometry.location.lat', 'geometry.location.lng', 'PostalCode', 'Street', 'city country', 'Fläche\nin km²_PLZ', 'Einwohner \nDez. 2018_PLZ', 'Einwohner\nDez. 2013_PLZ', 'Einwohner\nDez. 2005_PLZ', 'Ausländer \nDez. 2018_PLZ', 'Stadtkreis', 'KreisNumber', 'Fläche\nin km²_Kreis', 'Einwohner \nDez. 2018_Kreis', 'Einwohner\nDez. 2013_Kreis', 'Einwohner\nDez. 2005_Kreis', 'Ausländer \nDez. 2018_Kreis']
